internal/auth/nil_auth.go

```diff
@@ -2,16 +2,22 @@ package auth
 
 import (
 	"context"
+	"errors"
 	"net/http"
 )
 
 type NilAuth struct{}
 
-func (a NilAuth) AuthHandler(next http.Handler) http.Handler {
-	fn := func(w http.ResponseWriter, r *http.Request) {
-		next.ServeHTTP(w, r)
-	}
-	return http.HandlerFunc(fn)
+func (a NilAuth) AuthHandler(next http.Handler, w http.ResponseWriter, r *http.Request) {
+	next.ServeHTTP(w, r)
+}
+
+func (a NilAuth) ValidateToken(ctx context.Context) (bool, error) {
+	return false, errors.New("auth disabled")
```

Comment:
would it be useful to reuse might help for testing as well.

var ErrAuthDisabled = errors.New("auth disabled")

Reply:
done